app/src/main/java/com/sunasterisk/boringweather/util/ViewExtensions.kt

```diff
@@ -0,0 +1,24 @@
+package com.sunasterisk.boringweather.util
+
+import android.annotation.SuppressLint
+import androidx.core.widget.NestedScrollView
+import androidx.recyclerview.widget.RecyclerView
+
+val NestedScrollView.scrollProgress: Float
+    @SuppressLint("RestrictedApi")
+    get() {
+        val offset = computeVerticalScrollOffset()
+        val extent = computeVerticalScrollExtent()
+        val range = computeVerticalScrollRange()
+
+        return offset / (range - extent).toFloat()
+    }
+
+val RecyclerView.scrollProgress: Float
+    get() {
+        val offset = computeVerticalScrollOffset()
+        val extent = computeVerticalScrollExtent()
+        val range = computeVerticalScrollRange()
+
+        return offset / (range - extent).toFloat()
+    }
```

Comment:
cái này có thể dùng chung cho ScrollView và tên file nên để là ScrollView+.kt

Reply:
Viết thêm 1 cái cho scrollView à anh?

Reply:
Vì cả `NestedScrollView` và `RecyclerView đều implement interface `ScrollingView` nên em có thể làm như này
```kotlin
val ScrollingView.scrollProgress: Float
    get() {
        val offset = computeVerticalScrollOffset()
        val extent = computeVerticalScrollExtent()
        val range = computeVerticalScrollRange()

        return offset / (range - extent).toFloat()
    }
```
Đặt tên file là ScrollingView+.kt để dễ hiểu đây là extension của riêng ScrollingView chứ không phải của toàn bộ View

Reply:
Done